include/create/types.h

```diff
@@ -33,11 +33,41 @@ POSSIBILITY OF SUCH DAMAGE.
 #define CREATE_TYPES_H
 
 #include <vector>
+#include <stdint.h>
+#include <string>
+#include <stdexcept>
 
 namespace create {
-  enum RobotModel {
-    CREATE_1 = 0, // Roomba 400 series
-    CREATE_2      // Roomba 600 series
+  enum ProtocolVersion {
+    V_1 = 1,
+    V_2 = 2,
+    V_3 = 4,
+    V_ALL = 0xFFFFFFFF
+  };
+
+
+
+  class RobotModel {
+    public:
+      uint32_t id;
```

Comment:
Perhaps all of these member variables are best kept private and we use getters to access them, to enforce the use of the defined models below. What do you think?